extractor/src/main/java/org/schabi/newpipe/extractor/stream/StreamFrames.java

```diff
@@ -0,0 +1,113 @@
+package org.schabi.newpipe.extractor.stream;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class StreamFrames {
+
+	private final List<Frameset> frames;
+
+	public StreamFrames(String baseUrl, List<String> params) {
+		frames = new ArrayList<>(params.size());
```

Comment:
This link generation should be done in the YouTubeStreamExtractor above, since it is a youtube-specific thing. If, for example, PeerTube provides something similar, it wouldn't be possible to use this StreamFrames class.
Services implementations lie in the `extractor/services/` subfolder, and service-specific things should ONLY be there. All of the files not in that subfolder are: utilities common to every service (like `MediaFormat`), interfaces to be implemented by services (like `InfoItemExtractor`), interfaces to be implemented by NewPipe or users (like `Downloader`), classes wrapping around extracted data (like `InfoItem`).